File: javascript/chapter8/archived_messages.js
let allMessages = ["Hello there", "Readability counts", "Less is more", "Simple is better"];
let sentMessages = [];

sendMessages(allMessages, sentMessages);
console.log(allMessages);
console.log(sentMessages);


/**
 * @param {string[]} messages
 * @param {string[]} sent
 */
function sendMessages(messages, sent) {
	while (messages.length > 0) {
		sent.push(messages[0]);
		allMessages.shift();
	}
}

/** @param {string[]} messages */
function showMessages(messages) {
	for (const v in messages) console.log(v);
}

File: python/chapter8/archived_messages.py
all_messages = ["Hello there", "Readability counts", "Less is more", "Simple is better"]
sent_messages = list(str())


def send_messages(messages: list[str], sent: list[str]):
    while len(messages) > 0:
        sent.append(messages[0])
        all_messages.pop(0)


def show_messages(messages: list[str]):
    for msg in messages:
        print(msg)


send_messages(all_messages, sent_messages)
print(all_messages)
print(sent_messages)
